lib/wallet/api/http/Cardano/Wallet/Api/Http/Server/Error.hs

```diff
@@ -526,11 +526,15 @@ instance IsServerError ErrBalanceTx where
 
 instance IsServerError ErrBalanceTxInternalError where
     toServerError = \case
-        ErrUnderestimatedFee coin _st ->
+        ErrUnderestimatedFee coin candidateTx nWits ->
             apiError err500 BalanceTxUnderestimatedFee $ T.unwords
                 [ "I have somehow underestimated the fee of the transaction by"
                 , pretty coin
                 , "and cannot finish balancing."
+                , "I have assumed the transaction needs" <> T.pack (show nWits)
+                , "verification key witnesses.\n\n"
+                , "The candidate tx is:"
+                , T.pack (show candidateTx)
```

Comment:
Not a blocker for merging this PR, but:

I think it would be nice to make these API errors into rich JSON objects, rather than embedding all the information in a string.

For example:
https://github.com/input-output-hk/cardano-wallet/blob/43b0abd630e6e56c6cdd95a3906df0ee573ff704/lib/wallet/api/http/Cardano/Wallet/Api/Types/Error.hs#L162-L163
https://github.com/input-output-hk/cardano-wallet/blob/43b0abd630e6e56c6cdd95a3906df0ee573ff704/lib/wallet/api/http/Cardano/Wallet/Api/Types/Error.hs#L199-L209

Reply:
Feels borderline overkill in this case, but going with it:
```haskell
data ApiErrorBalanceTxUnderestimatedFee = ApiErrorBalanceTxUnderestimatedFee
    { underestimation :: !(Quantity "lovelace" Natural)
    , estimatedNumberOfKeyWits :: Natural
    , estimatedNumberOfBootstrapKeyWits :: Natural
    , candidateTxHex :: Text
    , candidateTxReadable :: Text
    }
    deriving (Data, Eq, Generic, Show, Typeable)
    deriving (FromJSON, ToJSON)
        via DefaultRecord ApiErrorBalanceTxUnderestimatedFee
    deriving anyclass NFData

```

```haskell
         ( Status
             { statusCode = 500
             , statusMessage = "Internal Server Error"
             }
         , Left
             ( ClientError
                 ( Object
                     ( fromList
                         [
                             ( "code"
                             , String "balance_tx_underestimated_fee"
                             )
                         ,
                             ( "info"
                             , Object
                                 ( fromList
                                     [
                                         ( "candidate_tx_hex"
                                         , String "84a40081825820bc6dd844fccbd9b0a984d7deb78ba74f7ddbf4696cceddfec339fabff35d9895000182a300581d714d72cf569a339a18a7d9302313983f56e0d96cd45bdcb1d6512dca6a011a001e84800282005820923918e403bf43c34b4ef6b48eb2ee04babed17320d8d1b9ff9ad086e86f44eca20058390158cc28a092e265d58139774342c0f378dc78dcb0c3b4fda37012ff4db0d4a8312a679abf89ed1acadd0164064765712d608aa90f1756385c011b000000174856a248021a0001aa2c0b58202f50ea2546f8ce020ca45bfcf2abeb02ff18af2283466f888ae489184b3d2d39a10481d87980f5f6"
                                         )
                                     ,
                                         ( "candidate_tx_readable"
                                         , String "SealedTx (InAnyCardanoEra BabbageEra (ShelleyTx ShelleyBasedEraBabbage (ValidatedTx {body = TxBodyConstr TxBodyRaw {_spendInputs = fromList [TxIn (TxId {_unTxId = SafeHash "bc6dd844fccbd9b0a984d7deb78ba74f7ddbf4696cceddfec339fabff35d9895"}) (TxIx 0)], _collateralInputs = fromList [], _referenceInputs = fromList [], _outputs = StrictSeq {fromStrict = fromList [Sized {sizedValue = (Addr Mainnet (ScriptHashObj (ScriptHash "4d72cf569a339a18a7d9302313983f56e0d96cd45bdcb1d6512dca6a")) StakeRefNull,Value 2000000 (fromList []),DatumHash (SafeHash "923918e403bf43c34b4ef6b48eb2ee04babed17320d8d1b9ff9ad086e86f44ec"),SNothing), sizedSize = 76},Sized {sizedValue = (Addr Mainnet (KeyHashObj (KeyHash "58cc28a092e265d58139774342c0f378dc78dcb0c3b4fda37012ff4d")) (StakeRefBase (KeyHashObj (KeyHash "b0d4a8312a679abf89ed1acadd0164064765712d608aa90f1756385c"))),Value 99997885000 (fromList []),NoDatum,SNothing), sizedSize = 71}]}, _collateralReturn = SNothing, _totalCollateral = SNothing, _certs = StrictSeq {fromStrict = fromList []}, _wdrls = Wdrl {unWdrl = fromList []}, _txfee = Coin 109100, _vldt = ValidityInterval {invalidBefore = SNothing, invalidHereafter = SNothing}, _update = SNothing, _reqSignerHashes = fromList [], _mint = Value 0 (fromList []), _scriptIntegrityHash = SJust (SafeHash "2f50ea2546f8ce020ca45bfcf2abeb02ff18af2283466f888ae489184b3d2d39"), _adHash = SNothing, _txnetworkid = SNothing}, wits = TxWitnessRaw {_txwitsVKey = fromList [], _txwitsBoot = fromList [], _txscripts = fromList [], _txdats = TxDatsRaw (fromList [(SafeHash "923918e403bf43c34b4ef6b48eb2ee04babed17320d8d1b9ff9ad086e86f44ec",DataConstr Constr 0 [])]), _txrdmrs = RedeemersRaw (fromList [])}, isValid = IsValid True, auxiliaryData = SNothing}))) "\132\164\NUL\129\130X \188m\216D\252\203\217\176\169\132\215\222\183\139\167O}\219\244il\206\221\254\195\&9\250\191\243]\152\149\NUL\SOH\130\163\NULX\GSqMr\207V\154\&3\154\CAN\167\217\&0#\DC3\152?V\224\217l\212[\220\177\214Q-\202j\SOH\SUB\NUL\RS\132\128\STX\130\NULX \146\&9\CAN\228\ETX\191C\195KN\246\180\142\178\238\EOT\186\190\209s \216\209\185\255\154\208\134\232oD\236\162\NULX9\SOHX\204(\160\146\226e\213\129\&9wCB\192\243x\220x\220\176\195\180\253\163p\DC2\255M\176\212\168\&1*g\154\191\137\237\SUB\202\221\SOHd\ACKGeq-`\138\169\SI\ETBV8\\\SOH\ESC\NUL\NUL\NUL\ETBHV\162H\STX\SUB\NUL\SOH\170,\vX /P\234%F\248\206\STX\f\164[\252\242\171\235\STX\255\CAN\175\"\131Fo\136\138\228\137\CANK=-9\161\EOT\129\216y\128\245\246" True"
                                         )
                                     ,
                                         ( "estimated_number_of_bootstrap_key_wits"
                                         , Number 0.0
                                         )
                                     ,
                                         ( "estimated_number_of_key_wits"
                                         , Number 1.0
                                         )
                                     ,
                                         ( "underestimation"
                                         , Object
                                             ( fromList
                                                 [
                                                     ( "quantity"
                                                     , Number 19000.0
                                                     )
                                                 ,
                                                     ( "unit"
                                                     , String "lovelace"
                                                     )
                                                 ]
                                             )
                                         )
                                     ]
                                 )
                             )
                         ,
                             ( "message"
                             , String "I have somehow underestimated the fee of the transaction by 0.019000 and cannot finish balancing."
                             )
                         ]
                     )
                 )
             )
         )
```